bindings/rust/generate/_generation_tools.sh

```diff
@@ -124,3 +124,52 @@ function create_prefix_headers {
     echo Prefix headers generation complete
   fi
 }
+
+function parse_version {
+  local VERSION="${1}"
+  echo Version: "${VERSION}"
+  echo "${VERSION}" | egrep -q '^[0-9]+\.[0-9]+\.[0-9]+$'
+}
+
+function determine_generate_version {
+  PUBLISHED_CRATE_VERSION=$(cargo search "${CRATE_NAME}" | egrep "^${CRATE_NAME} " | sed -e 's/.*"\(.*\)".*/\1/')
+
+  source "${SCRIPT_DIR}"/_generation_tools.sh
+
+  if ! parse_version "${PUBLISHED_CRATE_VERSION}"; then
+    echo Could not find current version of published crate.
+    exit 1
+  fi
+
+  while [ -z "${CRATE_VERSION}" ]; do
+    echo
+    echo Current published version of ${CRATE_NAME}: ${PUBLISHED_CRATE_VERSION}
+    read -p "Enter version for crate generation: " NEW_VERSION
+    if parse_version "${NEW_VERSION}"; then
+      CRATE_VERSION="${NEW_VERSION}"
+    else
```

Comment:
Probably should verify that it is higher then the previous version, that way we can catch data entry errors.